packages/jest-cli/src/reporters/coverage_reporter.js

```diff
@@ -224,10 +225,59 @@ class CoverageReporter extends BaseReporter {
           return errors;
         }, []);
       }
-      const errors = check(
-        'global',
-        globalConfig.coverageThreshold.global,
-        results,
+
+      const thresholdConfigFilePaths = Object.keys(
+        globalConfig.coverageThreshold,
+      )
+        .filter(thresholdKey => thresholdKey !== 'global')
+        .map(thresholdKey => path.resolve(thresholdKey));
+      const filteredCoverageSummary = map
+        .files()
+        .filter(filePath => thresholdConfigFilePaths.indexOf(filePath) === -1)
+        .map(filePath => map.fileCoverageFor(filePath))
+        .reduce((summary: ?CoverageSummary, fileCov: FileCoverage) => {
+          if (summary !== undefined && summary !== null) {
+            summary.merge(fileCov.toSummary());
+          } else {
+            summary = fileCov.toSummary();
+          }
+          return summary;
+        }, undefined);
+
+      const errors = [].concat(
+        ...Object.keys(globalConfig.coverageThreshold)
+          .map(thresholdKey => {
+            if (thresholdKey === 'global') {
+              if (filteredCoverageSummary !== undefined) {
+                return check(
+                  'global',
+                  globalConfig.coverageThreshold.global,
+                  filteredCoverageSummary,
+                );
+              } else {
+                return [];
+              }
+            } else {
+              if (map.files().indexOf(path.resolve(thresholdKey)) !== -1) {
+                return check(
+                  thresholdKey,
+                  globalConfig.coverageThreshold[thresholdKey],
+                  map.fileCoverageFor(path.resolve(thresholdKey)).toSummary(),
+                );
+              } else {
+                return [
+                  `Jest: Coverage data for ${thresholdKey} was not found.`,
+                ];
```

Comment:
I like `check` function being unaware of such details and just sticking to comparing numbers. I can add move this to `check` as a two distinct conditions if `actuals` or `thresholds` do not contain given key respectively. What do you think?